src/Command/PelagosMigrateDatasetFilesCommand.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace App\Command;
+
+use App\Entity\Dataset;
+use App\Entity\DatasetSubmission;
+use App\Entity\File;
+use App\Entity\FileSet;
+use App\Message\DatasetSubmissionFiler;
+
+use Doctrine\ORM\EntityManagerInterface;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+class PelagosMigrateDatasetFilesCommand extends Command
+{
+    protected static $defaultName = 'pelagos:migrate-dataset-files';
+
+    /**
+     * A Doctrine ORM EntityManager instance.
+     *
+     * @var EntityManagerInterface $entityManager
+     */
+    protected $entityManager;
+
+    /**
+     * The message bus for dispatching the filer message.
+     *
+     * @var MessageBusInterface $messageBus
+     */
+    protected $messageBus;
+
+    /**
+     * This contains a list of submission IDs that need to be queued to filer.
+     *
+     * @var Array $submissionsListForFiler
+     */
+    protected $submissionsListForFiler;
+
+    /**
+     * Class constructor for dependency injection.
+     *
+     * @param EntityManagerInterface $entityManager A Doctrine EntityManager.
+     */
+    public function __construct(EntityManagerInterface $entityManager, MessageBusInterface $messageBus)
+    {
+        $this->submissionsListForFiler = array();
```

Comment:
This can be initialized when it is declared on line 45, as the scope of this variable is global